### DISCUSSION TITLE
investpy not pulling data, Connection Error 403

### DISCUSSION THREAD

**Discussion by omarequalmars** [alvarobartt/investpy#622] [Q&A]
Hello, I just recently got to know investpy and I tried using the example code presented, it gave me this error and I looked it up and I also tried investiny but that gave me the same error. I saw a discussion from last september mentioning that investpy is being fixed because investing.com changed their API. I wanted to ask, is there hope for this library's revival or its sister investiny's revival?

![Screenshot 2023-02-20 214216](https://user-images.githubusercontent.com/90857121/220187392-f686d411-b655-4035-a73b-a5bd8d3048eb.png)

**Comment by alextaf7** [comment]
Any updates on this issue ?

**Comment by piyushraj7890** [comment]
Hi any update on this issue , as i am finding it hard to pull data from investpy

**Comment by ymyke** [comment]
There won't be a fix for this as investing.com made scraping much harder/impossible.

I recommend using yfinance instead – or my package https://github.com/ymyke/tessa, which provides easy access to both yfinance and pycoingecko (for crypto).
